js/ad_manager_vast.js

```diff
@@ -3630,62 +3400,331 @@ OO.Ads.manager(() => {
     };
 
     /**
-     * Utility function to check if an ad is a VPAID ad.
-     * @private
-     * @method Vast#_isVpaidAd
-     * @param ad The ad to check
-     * @returns {boolean}
+     * Initialize the vast ad manager. For the Vast Ad Manager it will set the local
+     * variable for amc to the current ad manager controller passed into the function and add the
+     * event listeners to the amc controller that is needed to handle positioning of the overlay when controls are shown
+     * and hidden. It will also add event Listeners that are needed for when the user goes into full screen mode or
+     * comes out of it.
+     * @public
+     * @method Vast#initialize
+     * @param {object} amc The current instance of the ad manager controller
      */
-    var _isVpaidAd = (ad) => {
-      const vastAdObject = _getVastAdObject(ad);
-      return vastAdObject.adType === 'vpaid';
+    this.initialize = (amc) => {
+      this.amc = amc;
+      this.amc.addPlayerListener(this.amc.EVENTS.INITIAL_PLAY_REQUESTED, this.initialPlay);
+      this.amc.addPlayerListener(this.amc.EVENTS.REPLAY_REQUESTED, this.replay);
+      this.amc.addPlayerListener(this.amc.EVENTS.FULLSCREEN_CHANGED, _onFullscreenChanged);
+      this.amc.addPlayerListener(this.amc.EVENTS.SIZE_CHANGED, _onSizeChanged);
+      this.amc.addPlayerListener(this.amc.EVENTS.AD_PLAYHEAD_TIME_CHANGED, this.onAdPlayheadTimeChanged);
+      this.amc.addPlayerListener(this.amc.EVENTS.AD_VOLUME_CHANGED, this.onAdVolumeChanged);
     };
 
-    // Helpers
-    // Safely trigger an ad manager function
-    // TODO: consider error message override
-    var _safeFunctionCall = (vpaidAd, funcName, params) => {
-      try {
-        if (isFunction(vpaidAd[funcName])) {
-          return vpaidAd[funcName].apply(vpaidAd, params);
+    /**
+     * Callback when the media file is loaded. Once is loaded we can initialize the ad
+     * This is only required for VPAID ads
+     * @public
+     * @returns {null} Returns null
```

Comment:
It shouldn't return anything, just `void`. Please correct `return null;` to just `return;`